// SPDX-FileCopyrightText: 2025 Carnegie Mellon University - Living Edge Lab
//
// SPDX-License-Identifier: GPL-2.0-only

syntax = "proto3";
package protocol.testing;

/*
 * Types of test messages for testing infrastructure
 */
enum ServiceType {
  CONTROL_PROXY_SERVICE = 0;
  REPORT_SERVICE = 1;
  SC_HANDLER_SERVICE = 2;
  COMPUTE_SERVICE = 3; 
  DRIVER_CONTROL_SERVICE = 4;
}

message TestStart {
  int32 test_id = 1;
}

message ServiceReady {
  // Indicates which service is ready for testing
  ServiceType readied_service = 1;
}
